.github/workflows/release.yaml

```diff
@@ -6,27 +6,74 @@ on:
 env:
   go-version: 1.17
 jobs:
-  image:
+  build-moco:
     name: Push Container Image
     runs-on: ubuntu-20.04
     steps:
     - uses: actions/checkout@v3
-    - name: Build images
-      run: |
-        docker build -t moco:dev --target controller .
-        docker build -t moco-backup:dev .
-    - name: Login to ghcr.io
-      run: echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
-    - name: Push images
-      run: |
-        TAG=${GITHUB_REF#refs/tags/v}
-        docker tag moco:dev ghcr.io/cybozu-go/moco:$TAG
-        docker push ghcr.io/cybozu-go/moco:$TAG
-        docker tag moco-backup:dev ghcr.io/cybozu-go/moco-backup:$TAG
-        docker push ghcr.io/cybozu-go/moco-backup:$TAG
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v4
+      with:
+        images: ghcr.io/cybozu-go/moco
+        # ref. https://github.com/docker/metadata-action#typesemver
+        tags: |
+          type=semver,pattern={{version}}
+    - name: Build (and push if on main)
+      id: docker_build
+      uses: docker/build-push-action@v2
+      with:
+        builder: ${{ steps.buildx.outputs.name }}
+        push: true
```

Comment:
You can also test docker build on each commit if you set this as false on other action triggers.